snippets/RGBToHSV.md

```diff
@@ -0,0 +1,44 @@
+---
+title: RGBToHSV
+tags: math, intermediate
+---
+
+Converts the values of RGB color components to HSV components.
+
+Convert given RGB parameters into HSV ones and returns it in array format. 
+```js
+const RGBToHSV =  (r, g, b) => {
+                const hsv = [];
+                r/=255.0; g/=255.0; b/=255.0;
+                const max = Math.max(r,g,b);
+                const min = Math.min(r,g,b);
+                const diff = max - min;
+
+                //computing h  
+                if(max === min){
+                    hsv.push(0);
+                }else if (max === r){
+                    hsv.push((60 * ((g - b) / diff) + 360) % 360);
+                }else if(max === g){
+                    hsv.push((60 * ((b - r) / diff) + 120) % 360);
+                }else{
+                    hsv.push((60 * ((r - g) / diff) + 240) % 360);
+                }
+
+                //computing s
+                if(max === 0){
+                    hsv.push(0);
+                }else{
+                    hsv.push((diff / max) * 100);
+                }
+
+                //computing v
+                hsv.push(max*100);
+                
+                return hsv;
+            };
```

Comment:
I am pretty sure this can be simplified down to:

```suggestion
const RGBToHSV =  (r, g, b) => {
  r /= 255; g /= 255; b /= 255;
  const v = Math.max(r,g,b), n = v - Math.min(r,g,b);
  const h = n && v === r 
    ? (g - b) / n 
    : v === g 
      ? 2 + (b - r) / n 
      : 4 + (r - g) / n; 
  return [60 * (h < 0 ? h + 6 : h), v && n / v * 100, v * 100];
};
```